### PR TITLE
DSECT USS utility

### PR DESCRIPTION
**Pull request by MikeFultonDev**
Request to add in a new utility: edcdsect.sh which lets you generate C structures from assembler dsects.
Requires ASMA90 (HL/ASM), C/C++ DSECT utility, and of course Z Open Automation Utilities

### CODE REVIEW

File: samples/edcdsect.sh [hunk 1]
```diff
@@ -0,0 +1,110 @@
+#!/bin/env bash
+
+# 
+# EDCDSECT USS invocation. 'edcdsect -?' for help (see below)
```
Comment (AnthonyGiorgio) [inline samples/edcdsect.sh:4]: I prefer `-h` for the help option.

File: samples/edcdsect.sh [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+Options:
+  -h, --help        display this help and exit.
+  -v, --verbose     run in verbose mode.
+
+Parameters:
+  <file>            The assembler file to process
+
+Examples:
+  Process a single macro DCBD into a structure:
+    echo '      DCBD' | edcdsect
```
Comment (AnthonyGiorgio) [inline samples/edcdsect.sh:20]: Why all the blank spaces to the left of `DCBD`?

File: samples/edcdsect.sh [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+
+tmp_output=$(mvstmp $(hlq))
+if ! dtouch -rvb -l137 -tseq "${tmp_output}" ; then
+  printf 'Unable to allocate DSECT temporary output dataset %s' "${tmp_output}"
+  exit 4
+fi
+err=$(mvscmd --pgm=CCNEDSCT --args="SECT(${struct_name}),EQU,NODEF,UNNAMED" --steplib="${compiler}" --sysadata="${sysadata}" --edcdsect="${tmp_output}" --sysprint=stdout --sysout=stdout)
+rc=$?
+
+cat "//'${tmp_output}'"
```
Comment (AnthonyGiorgio) [inline samples/edcdsect.sh:99]: Why not use `dcat`?

File: samples/edcdsect.sh [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+  done </dev/stdin
+  IFS="$OLDIFS"
+fi
+
+compiler="CBC.SCCNCMP"
+
+asm='ASMA90'
+asm_opts='SUPRWARN(425,434),GOFF,ADATA,NOTERM,NODECK,NOOBJECT,LIST'
+sysadata=$(mvstmp $(hlq))
+if ! dtouch -rvb -l8144 -tseq "${sysadata}" ; then
```
[the remark this answers is not in the document]
Comment (MikeFultonDev) [reply samples/edcdsect.sh:74]: ok - i would have thought mvstmp would make this virtually impossible?

[the remark this answers is not in the document]
Comment (AnthonyGiorgio) [reply samples/edcdsect.sh:77]: `mvstmp` doesn't check for existence.  It just generates a dataset name based on a few inputs.  Perhaps we need a more robust solution that creates one atomically.  Something like `tmpnam` but for datasets.

**Review by AnthonyGiorgio — APPROVED** [review]

**Pull request merged**